### PR TITLE
WIP: GPU-compatibility of Jacobians

### PR DESCRIPTION
**Pull request by ChrisRackauckas**
MWE:

```julia
using ForwardDiff
A = cu(-rand(3,3))
u = cu([1.0;0.0;0.0])
du = similar(u)
f(du,u) = mul!(du,A,u)
ForwardDiff.jacobian(f,du,u)
```

### CODE REVIEW

**Comment by ChrisRackauckas** [comment]
@vchuravy any idea why this would fallback to the Base broadcasting?

```julia
using ForwardDiff, CuArrays
x = CuArray([ForwardDiff.Dual(1.0f0,(0.0f0,0.0f0)) for i in 1:10])
CuArrays.allowscalar(false)
ForwardDiff.partials.(Float32,x,1)
```

**Comment by KristofferC** [comment]
https://github.com/JuliaDiff/ForwardDiff.jl/pull/351

The reason that PR got reverted needs to be checked here.

**Comment by ChrisRackauckas** [comment]
I see. As a continuation of https://github.com/JuliaDiff/ForwardDiff.jl/pull/353 @maleadt . The routes I'm thinking of are:

1) Requires.jl, add overloads for CuArrays. Would you accept that?
2) We may just add overloads for this in Requires of DiffEqBase. DiffEqBase has a faster `@..` broadcast which gets rid of the timing issues of changing to broadcast (Julia v1.0 broadcast is quite slow :( ), and I'm not sure we could have no performance loss without using this.
3) We may just make sure the color-autodiff form of this into SparseDiffTools.jl.

Of course, not having it in ForwardDiff.jl is a hassle, and having the piracy version in DiffEqBase is a little awkward, but my main priority is to get it working for DiffEq and then find out where to put the code later.

File: src/apiutils.jl [hunk 1]
```diff
@@ -55,36 +55,31 @@ end
...
     return duals
 end
 
 function seed!(duals::AbstractArray{Dual{T,V,N}}, x,
                seeds::NTuple{N,Partials{N,V}}) where {T,V,N}
-    for i in 1:N
-        duals[i] = Dual{T,V,N}(x[i], seeds[i])
-    end
+    i = 1:N
+    @views duals[i] .= Dual{T,V,N}.(x[i], seeds)
```
Comment (KristofferC) [inline src/apiutils.jl:65]: Perhaps these `@views` could be reduced in "strength" by putting them where they are needed, e.g. only on the `x[i]`? And could the usage of `@.` be consistent among the changes (it's not used here, but used in other places).

**Comment by charleskawczynski** [comment]
I think we can close this now that #472 has merged.

**Pull request closed without merging**